### PR TITLE
additional message highlighting - os notifications, unread messages, improvements

### PR DESCRIPTION
**Pull request by Mashintime**
- Call list will have an '*' prefixed to call sign and background color set if a new message is available
- Show notifications via the operating system on CQ and messages; configuration option available to disable
- RF Chat button background will change to yellow to indicate a new message/activity is available
- Minor console.log clean up in chat module
- Database pruning option added to clean older pings and beacons out of the database
- Ignore unusable flex radio audio devices (there are just too many)

Ready for testing and feedback.

### CODE REVIEW

File: gui/main.js [hunk 1]
```diff
@@ -100,7 +100,8 @@ const configDefaultSettings =
                   "max_retry_attempts" : 5, \
                   "enable_auto_retry" : "False", \
                   "tx_delay" : 0, \
-                  "auto_start": 0 \
+                  "auto_start": 0, \
+                  "notification": 1 \
```
Comment (DJ2LS) [inline gui/main.js:104]: Can we name this more self explaining like "enable_system_notification" or something like that?

File: tnc/audio.py [hunk 2]
```diff
@@ -76,6 +76,9 @@ def fetch_audio_devices(input_devices, output_devices):
         # Use a try/except block because Windows doesn't have an audio device range
         try:
             name = device["name"]
+            # Ignore some Flex Radio devices to make device selection simpler
+            if name.startswith("DAX RESERVED") or name.startswith("DAX IQ"):
```
Comment (DJ2LS) [inline tnc/audio.py:80]: Nice improvement 🙂 Thanks!

**Review by DJ2LS — COMMENTED** [review]
Nice improvements! Question: Can we count unread messages and displaying this somehow? Also instead of just using "*" can we use a nice gui element or badge as well?

File: gui/main.js (hunk 1 above)
Comment (Mashintime) [reply gui/main.js]: Updated to enable_sys_notification.

File: tnc/audio.py (hunk 2 above)
Comment (Mashintime) [reply tnc/audio.py:80]: Quality of life improvements are always a plus.

**Comment by Mashintime** [comment]
> Nice improvements! Question: Can we count unread messages and displaying this somehow? Also instead of just using "*" can we use a nice gui element or badge as well?

We could count unread messages, but personally feel it's over kill.  I feel just knowing there are new messages is enough.  Plus no real further GUI elements to add or jumble up the interface is a plus in my opinion.

The '*' was just a lazy/quick implementation.  The callsign area is already so busy, but if you have an idea let me know.

**Comment by DJ2LS** [comment]
We could use it as it is for now and if we have some users requesting this we could think again about it.

**Comment by Mashintime** [comment]
The * should clear on first click.  If you get an error in console let me know.  The chat button doesn't update immediately and is delayed to limit cpu load.

**Comment by DJ2LS** [comment]
@Mashintime ah its working! I didnt see there was another unread message....

**Comment by DJ2LS** [comment]
@Mashintime After some days of testing I would suggest, that a beacon or a ping isnt triggering "unread" message flag. Only messages should mark a chat as "unread".  I'm also wondering about sort oder of unread messages and maybe we can make the background a different color for additional highlighting

**Comment by Mashintime** [comment]
I'll remove the new flag from ping and beacon, easy enough.  Like the idea for changing background color.  Should we do background and * or do you think background would be enough.

I'll look at sort order again, but really won't matter unless user refreshes.

**Comment by DJ2LS** [comment]
I think both is fine. Then we have a text indicator and a colour indicator as well.

**Comment by Mashintime** [comment]
@DJ2LS feel this is ready for merging after your review.

**Comment by DJ2LS** [comment]
@Mashintime looks good so far, question:
is the delete database button just for the selected chat or for the entire database? If so, I suggest, we should add it to app settings instead to chat dropdown.

**Comment by DJ2LS** [comment]
And I think we could add pings to cleanups as well

**Comment by DJ2LS** [comment]
Oh, and @Mashintime , can you update bootstrap from alpha to 5.3.0 ? We have a stable release now

**Comment by Mashintime** [comment]
> Oh, and @Mashintime , can you update bootstrap from alpha to 5.3.0 ? We have a stable release now

No problem

> And I think we could add pings to cleanups as well

Will do

> is the delete database button just for the selected chat or for the entire database? If so, I suggest, we should add it to app settings instead to chat dropdown.

It's for the entire database.  Will move to settings.

**Comment by DJ2LS** [comment]
> > Oh, and @Mashintime , can you update bootstrap from alpha to 5.3.0 ? We have a stable release now
> 
> No problem
> 
> > And I think we could add pings to cleanups as well
> 
> Will do
> 
> > is the delete database button just for the selected chat or for the entire database? If so, I suggest, we should add it to app settings instead to chat dropdown.
> 
> It's for the entire database. Will move to settings.

I think then we can merge this

**Review by DJ2LS — APPROVED** [review]
Looks good so far, great work!

**Pull request merged**